pkg/controller/nodes/task/transformer.go

```diff
@@ -69,55 +69,73 @@ func getParentNodeExecIDForTask(taskExecID *core.TaskExecutionIdentifier, execCo
 	return nodeExecutionID, nil
 }
 
-func ToTaskExecutionEvent(taskExecID *core.TaskExecutionIdentifier, in io.InputFilePaths, out io.OutputFilePaths, info pluginCore.PhaseInfo,
-	nodeExecutionMetadata handler.NodeExecutionMetadata, execContext executors.ExecutionContext) (*event.TaskExecutionEvent, error) {
+type ToTaskExecutionEventInputs struct {
+	TaskExecID            *core.TaskExecutionIdentifier
+	InputReader           io.InputFilePaths
+	OutputWriter          io.OutputFilePaths
+	Info                  pluginCore.PhaseInfo
+	NodeExecutionMetadata handler.NodeExecutionMetadata
+	ExecContext           executors.ExecutionContext
+	TaskType              string
+	PluginID              string
+}
+
+func ToTaskExecutionEvent(input ToTaskExecutionEventInputs) (*event.TaskExecutionEvent, error) {
 	// Transitions to a new phase
 
 	tm := ptypes.TimestampNow()
 	var err error
-	if i := info.Info(); i != nil && i.OccurredAt != nil {
+	if i := input.Info.Info(); i != nil && i.OccurredAt != nil {
 		tm, err = ptypes.TimestampProto(*i.OccurredAt)
 		if err != nil {
 			return nil, err
 		}
 	}
 
-	nodeExecutionID, err := getParentNodeExecIDForTask(taskExecID, execContext)
+	nodeExecutionID, err := getParentNodeExecIDForTask(input.TaskExecID, input.ExecContext)
 	if err != nil {
 		return nil, err
 	}
+	metadata := input.Info.Info().Metadata
```

Comment:
We should populate Metadata with all the things we know/control in propeller... GeneratedName, ResourcePoolInfo... (etc.?)

Reply:
done, PTAL